File: algorithms/594. Longest Harmonious Subsequence/solution.cc
class Solution {
public:
    int findLHS(vector<int>& nums) {
        map<int, int> counter;
        for ( int n : nums ) {
            ++counter[n];
        }
        int lhs = 0;
        for ( auto & p : counter ) {
            if ( counter.find(p.first + 1) != counter.end() ) {
                lhs = max(lhs, p.second + counter[p.first + 1]);
            }
        }
        return lhs;
    }
};

File: algorithms/594. Longest Harmonious Subsequence/solution.py
class Solution(object):
    def findLHS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        counter = collections.Counter(nums)
        lhs = 0
        for n in counter:
            if n + 1 in counter:
                lhs = max(lhs, counter[n] + counter[n+1])
        return lhs
